addons/auth_signup_recaptcha/controllers/signup.py

```diff
@@ -0,0 +1,18 @@
+from odoo import http, _
+from odoo.addons.auth_signup.controllers.main import AuthSignupHome
+from odoo.http import request
+from odoo.exceptions import UserError
+
+
+class SignupCaptcha(AuthSignupHome):
+    @http.route()
+    def web_auth_signup(self, *args, **kw):
+        if request.httprequest.method == 'POST' and not request.env['ir.http']._verify_request_recaptcha_token('portal_signup'):
+            raise UserError(_("Suspicious activity detected by Google reCaptcha."))
+        return super().web_auth_signup(*args, **kw)
+
+    @http.route()
+    def web_auth_reset_password(self, *args, **kw):
+        if request.httprequest.method == 'POST' and not request.env['ir.http']._verify_request_recaptcha_token('portal_signup'):
+            raise UserError(_("Suspicious activity detected by Google reCaptcha."))
+        return super().web_auth_reset_password(*args, **kw)
```

Comment:
I hate/avoid overriding routes as much as possible.

+ This is copy/pasted code for each signup and reset password
+ the same protection must also be added on login, to counter credential stuffing attacks, as done on odoo.com.
+ maybe we could imagine a generic mechanism to be able to easily add the protection using an additional parameter to `@route`, like `csrf=True`, but this one would be on-demand. Such as `recaptcha=True`. 

Wouldn't it be better to create a factorization method in `ir.http` ?
- Either which gets called during the HTTP dispatching when `recaptcha=True` is set on the route
- Either we manually add a call to this factorized in the original login/signup/reset password routes
  - in base, the factorized wouldn't do anything,
  - in this module, it would override this method to add the token verification

That would avoid to have to override controller, and avoid a lot of copy/pasting.
This is debatable though.